### PR TITLE
KZG Initialization Refactor

### PR DESCRIPTION
**Pull request by holgerd77**
Small refactor in preparation for a simplified KZG initialization on the EthereumJS side as discussed in the chat.

### CODE REVIEW

**Comment by acolytec3** [comment]
Have revised this slightly as follows:
- Removed setup path from `loadKZG`
- Added optional param to `loadTrustedSetup` with trusted setup dictionary type
- Loads mainnet trusted setup if alternative not provided

File: src/index.ts [hunk 1]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
     const verifyBlobKzgProofWasm = module.cwrap('verify_blob_kzg_proof_wasm', 'string', ['array', 'array', 'array']) as (blob: Uint8Array, commitment: Uint8Array, proof: Uint8Array) => string
     const verifyKzgProofWasm = module.cwrap('verify_kzg_proof_wasm', 'string', ['array', 'array', 'array', 'array']) 
 
-    const loadTrustedSetup = (trustedSetup: TrustedSetup) => {
+    /**
+     * 
+     * @param trustedSetup - an optional trusted setup parameter provided by the user
+     * @returns 0 if loaded successfully or 1 otherwise
+     */
+    const loadTrustedSetup = (trustedSetup?: TrustedSetup) => {
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
    const loadTrustedSetup = (trustedSetup: TrustedSetup = mainnetTrustedSetup) => {
```
small nit, and then we can remove the undefined check

File: src/index.ts [hunk 2]
```diff
@@ -9,9 +9,12 @@ export type TrustedSetup = {
 }
 /**
  * Initialization function that instantiates WASM code and returns an object matching the `KZG` interface exposed by `@ethereumjs/util`
+ * 
+ * @param setupPath Optional setup path, otherwise official KZG setup from the KZG ceremony is used
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
 * @param setupPath Optional setup, otherwise official KZG setup from the KZG ceremony is used
```

**Review by acolytec3 — APPROVED** [review]
LGTM.  WIll merge once @g11tech approves

File: src/index.ts [hunk 3]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+    const loadTrustedSetup = (trustedSetup?: TrustedSetup) => {
+        if (trustedSetup === undefined) {
+            trustedSetup = mainnetTrustedSetup
+        }
         return loadTrustedSetupWasm(trustedSetup.g1, trustedSetup.n1, trustedSetup.g2, trustedSetup.n2)
     }
     
+    /**
+     * 
+     * @param blob - a blob of data formatted as a Uint8Array of 4096 bytes
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
     * @param blob - a blob of data formatted as a flattened Uint8Array of 4096 big endian KZG field elements (32 bytes)
```
1 blob is around 160kb i think

File: src/index.ts [hunk 4]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+        if (trustedSetup === undefined) {
+            trustedSetup = mainnetTrustedSetup
+        }
         return loadTrustedSetupWasm(trustedSetup.g1, trustedSetup.n1, trustedSetup.g2, trustedSetup.n2)
     }
     
+    /**
+     * 
+     * @param blob - a blob of data formatted as a Uint8Array of 4096 bytes 
+     * @returns a KZG commitment corresponding to the input blob formatted as a 32 byte Uint8Array
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
     * @returns a KZG commitment corresponding to the input blob formatted as a 48 byte Uint8Array
```
commitment is 48 bytes

File: src/index.ts [hunk 5]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+     */
     const blobToKzgCommitment = (blob: Uint8Array) => {
         const blobHex = '0x' + blobToKzgCommitmentWasm(blob)
         return hexToBytes(blobHex)
     }
 
+    /**
+     * 
+     * @param blob  - a blob of data formatted as a Uint8Array of 4096 bytes
```
Comment (g11tech) [inline src/index.ts]: again correction needed for what is blob and size

File: src/index.ts [hunk 6]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+     */
     const blobToKzgCommitment = (blob: Uint8Array) => {
         const blobHex = '0x' + blobToKzgCommitmentWasm(blob)
         return hexToBytes(blobHex)
     }
 
+    /**
+     * 
+     * @param blob  - a blob of data formatted as a Uint8Array of 4096 bytes 
+     * @param commitment - a KZG commitment corresponding to a blob formatted as a 32 byte Uint8Array
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
     * @param commitment - a KZG commitment corresponding to a blob formatted as a 48 byte Uint8Array
```

File: src/index.ts [hunk 7]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+     * @returns a KZG proof corresponding to the blob and KZG commitment
```
Comment (g11tech) [inline src/index.ts]:
```suggestion
     * @returns a 48 bytes KZG proof corresponding to the blob and KZG commitment
```

File: src/index.ts [hunk 8]
```diff
@@ -21,20 +24,46 @@ export const createKZG = async () => {
...
+     * @returns a KZG commitment corresponding to the input blob formatted as a 32 byte Uint8Array
+     */
     const blobToKzgCommitment = (blob: Uint8Array) => {
         const blobHex = '0x' + blobToKzgCommitmentWasm(blob)
         return hexToBytes(blobHex)
     }
 
+    /**
+     * 
+     * @param blob  - a blob of data formatted as a Uint8Array of 4096 bytes
```
Comment (acolytec3) [inline src/index.ts]:
```suggestion
     * @param blob  - a blob of data formatted as a flattened Uint8Array of 4096 big endian KZG field elements
```

File: src/index.ts [hunk 9]
```diff
@@ -46,15 +75,30 @@ export const createKZG = async () => {
         return true
     }
 
+    /**
+     * 
+     * @param blob - a blob of data formatted as a 4096 byte Uint8Array
```
Comment (g11tech) [inline src/index.ts]: same correction for blob/size

**Review by acolytec3 — COMMENTED** [review]
Have also updated the changelog and release so we can push an update on NPM.

**Pull request merged**